api/src/main/java/org/openmrs/module/addresshierarchy/service/AddressHierarchyServiceImpl.java

```diff
@@ -705,18 +705,26 @@ else if (level.getParent() == null){
 	synchronized public void initializeFullAddressCache() {
 
 		// generally, this global property should be set to true; it just allows cache load to be disabled to speed startup
-		if (Context.getAdministrationService().getGlobalProperty(AddressHierarchyConstants.GLOBAL_PROP_INITIALIZE_ADDRESS_HIERARCHY_CACHE_ON_STARTUP).equalsIgnoreCase("true")) {
-
-			// only initialize if necessary (and if we have entries)
-			if ((this.fullAddressCacheInitialized == false || MapUtils.isEmpty(this.fullAddressCache))
-					&& this.getAddressHierarchyEntryCount() > 0) {
-
-				this.fullAddressCache = new HashMap<Locale, Map<String,List<String>> >();
-				Locale locale = i18nCache.getLocaleForFullAddressCache();
-				getAddressesForLocale(locale);
-				this.fullAddressCacheInitialized = true;
+		try {
+			Context.addProxyPrivilege("Get Global Properties");

```

Comment:
I would use a locally declared constant for the above string.

Reply:
fine. Let me do that!

Reply:
> I would use a locally declared constant for the above string.

Hi @dkayiwa, I am done with this!